plasmapy/particles/tests/test_particle_class.py

```diff
@@ -995,3 +998,113 @@ def test_customized_particle_repr(cls, kwargs, expected_repr):
             f"from_str: {from_str}"
             f"from_repr: {from_repr}"
         )
+
+
+custom_particles_from_json_tests = [
+    (
+        DimensionlessParticle,
+        {"mass": 5.2, "charge": 6.3},
+        '{"plasmapy_particle": {"type": "DimensionlessParticle", "module": "plasmapy.particles.particle_class", "date_created": "2020-06-08", "init_args": {"mass": "5.2", "charge": "6.3"}}}',
+    ),
+    (
+        DimensionlessParticle,
+        {"mass": 5.2},
+        '{"plasmapy_particle": {"type": "DimensionlessParticle", "module": "plasmapy.particles.particle_class", "date_created": "2020-06-08", "init_args": {"mass": "5.2", "charge": "nan"}}}',
+    ),
+    (
+        CustomParticle,
+        {"mass": 5.12 * u.kg, "charge": 6.2 * u.C},
+        '{"plasmapy_particle": {"type": "CustomParticle", "module": "plasmapy.particles.particle_class", "date_created": "2020-06-08", "init_args": {"mass": "5.12 kg", "charge": "6.2 C"}}}',
+    ),
+    (
+        CustomParticle,
+        {"mass": 5.12 * u.kg},
+        '{"plasmapy_particle": {"type": "CustomParticle", "module": "plasmapy.particles.particle_class", "date_created": "2020-06-08", "init_args": {"mass": "5.12 kg", "charge": "nan C"}}}',
+    ),
+]
```

Comment:
```suggestion
custom_particles_from_json_tests = [
    (
        DimensionlessParticle,
        {"mass": 5.2, "charge": 6.3},
        '{"plasmapy_particle": {"type": "DimensionlessParticle", 
        "module": "plasmapy.particles.particle_class", 
        "date_created": "2020-06-08", 
        "__init__": {"args": [], "kwargs": {"mass": 5.2, "charge": 6.3}}}}',
    ),
    (
        DimensionlessParticle,
        {"mass": 5.2},
        '{"plasmapy_particle": {"type": "DimensionlessParticle", 
        "module": "plasmapy.particles.particle_class", 
        "date_created": "2020-06-08", 
        "__init__": {"args": [], "kwargs": {"mass": 5.2, "charge": NaN}}}}',
    ),
    (
        CustomParticle,
        {"mass": 5.12 * u.kg, "charge": 6.2 * u.C},
        '{"plasmapy_particle": {"type": "CustomParticle", 
        "module": "plasmapy.particles.particle_class", 
        "date_created": "2020-06-08", 
        "__init__": {"args": [], "kwargs": {"mass": "5.12 kg", "charge": "6.2 C"}}}}',
    ),
    (
        CustomParticle,
        {"mass": 5.12 * u.kg},
        '{"plasmapy_particle": {"type": "CustomParticle", 
        "module": "plasmapy.particles.particle_class", 
        "date_created": "2020-06-08", 
        "__init__": {"args": [], "kwargs": {"mass": "5.12 kg", "charge": "nan C"}}}}',
    ),
]
```

Reply:
I tried this one out, but ended up with an `EOL while scanning string literal` error. Apparently these strings can't be split across multiple lines? Not very sure about this one.

Reply:
It should work just fine.  I tested all these suggestions before suggesting them.

Note: You don't have to hand copy these suggestions to your local code.  You can directly commit the suggestion to your branch from the PR.  You can commit one suggestion at a time by using the "Commit suggestion" button.  Or, you can commit a batch of suggestions by going to the "Files changed" tab and add each suggest to a batch with the "Add suggestion to batch" button.

Reply:
Oh I see. Thank you for the suggestion. I wasn't aware of this feature. Definitely convenient.